addons/mail/static/src/js/activity.js

```diff
@@ -714,7 +714,7 @@ var KanbanActivity = BasicActivity.extend({
     _renderDropdown: function () {
         var self = this;
         this.$('.o_activity')
-            .toggleClass('dropdown-menu-right', config.device.isMobile)
+            .toggleClass('dropdown-menu-right', config.device.isMobile || self.$('.dropdown-toggle').offset()['left'] + self.$('.o_activity').outerWidth() > window.innerWidth)
```

Comment:
For some reason, the caracter just before `self` is a non-breaking space. You may want to replace it with a regular space.